src/app/main.cpp

```diff
@@ -192,12 +193,31 @@ int main(int argc, char *argv[])
             }
 #endif
 
+            if (!params.changeWebuiUsername.isEmpty() || !params.changeWebuiPassword.isEmpty())
+            {
+                throw CommandLineParameterError(QObject::tr("You cannot change the WebUI credentials "
+                                                            "via the commandline while qBittorrent is already running."));
+            }
+
             QThread::msleep(300);
             app->callMainInstance();
 
             return EXIT_SUCCESS;
         }
 
+        if (bool username = !params.changeWebuiUsername.isEmpty()
+            , password = !params.changeWebuiPassword.isEmpty(); username || password)
+        {
+            Preferences *const pref = Preferences::instance();
+            if (username)
+                pref->setWebUiUsername(params.changeWebuiUsername);
+
+            if (password)
+                pref->setWebUIPassword(Utils::Password::PBKDF2::generate(params.changeWebuiPassword));
+
+            return EXIT_SUCCESS;
```

Comment:
We don't exit app when Web UI port is changed via command line. Is there any strong reason to behave differently in this case?
If not, then I would also move username/password processing to Application class next to the WebUI port processing.